### PR TITLE
Make it possible to provide an upstream Theia repository through a command-line argument for building Che-Theia image

### PR DESCRIPTION
**Pull request by azatsarynnyy**
Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>

### What does this PR do?
Adds a command-line argument for passing an upstream Theia repository for building a Che-Theia image.
It would allow checking Che-Theia build against some Theia patch without changing Che-Theia sources.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=next

### CODE REVIEW

File: dockerfiles/theia/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,7 @@ WORKDIR ${HOME}
 ARG GITHUB_TOKEN=''
 ENV GITHUB_TOKEN=$GITHUB_TOKEN
 
-ARG THEIA_GITHUB_REPO=eclipse-theia/theia
+ARG THEIA_GITHUB_REPO=
```
Comment (benoitf) [inline dockerfiles/theia/Dockerfile:21]: what happens if we call build.sh from dockerfiles/theia folder ?

Comment (azatsarynnyy) [reply dockerfiles/theia/Dockerfile]: got it :slightly_smiling_face: - updated
Thanks!

**Review by benoitf — APPROVED** [review]

**Comment by azatsarynnyy** [comment]
[crw-ci-test]

**Comment by benoitf** [comment]
it passed on Happy Path / happy-path (alpine, next) (push)

**Pull request merged**